tests/udp/udp-netapi-tests.c

```diff
@@ -0,0 +1,295 @@
+    }
+
+    if (src != NULL) {
+        src_len = sizeof(ng_ipv6_addr_t);
+    }
+
+    if (dst != NULL) {
+        dst_len = sizeof(ng_ipv6_addr_t);
+    }
+
+    ipv6 = ng_ipv6_hdr_build(udp, (uint8_t *)src, src_len, (uint8_t *)dst, dst_len);
+
+    return ipv6;
+}
+
+static inline ng_pktsnip_t *build_ipv6_recv_header(const ng_ipv6_addr_t *src,
+        const ng_ipv6_addr_t *dst, size_t data_len)
+{
+    ng_pktsnip_t *ipv6;
+    size_t src_len = 0, dst_len = 0;
+    ng_ipv6_hdr_t *ipv6_hdr;
+
+    if (src != NULL) {
+        src_len = sizeof(ng_ipv6_addr_t);
+    }
+
+    if (dst != NULL) {
+        dst_len = sizeof(ng_ipv6_addr_t);
+    }
+
+    ipv6 = ng_ipv6_hdr_build(NULL, (uint8_t *)src, src_len, (uint8_t *)dst, dst_len);
+
+    if (ipv6 != NULL) {
+        ipv6_hdr = ipv6->data;
+        ipv6_hdr->hl = 0;
+        ipv6_hdr->nh = NG_PROTNUM_UDP;
+        ipv6_hdr->len = byteorder_htons(sizeof(ng_udp_hdr_t) + data_len);
+    }
+
+    return ipv6;
+}
+
+static ng_pktsnip_t *build_udp_recv_packet(const ng_ipv6_addr_t *src,
+        uint16_t src_port, const ng_ipv6_addr_t *dst, uint16_t dst_port,
+        uint16_t checksum, char *data, size_t data_len)
+{
+    ng_pktsnip_t *ipv6, *payload;
+    ng_udp_hdr_t *udp_hdr;
+    uint8_t *payload_data;
+
+    ipv6 = build_ipv6_recv_header(src, dst, data_len);
+
+    if (ipv6 == NULL) {
+        return NULL;
+    }
+
+    payload = ng_pktbuf_add(ipv6, NULL, sizeof(ng_udp_hdr_t) + data_len,
+                            NG_NETTYPE_UDP);
+
+    if (payload == NULL) {
+        /* no need to release, we reset packet buffer on fail anyways */
+        return NULL;
+    }
+
+    udp_hdr = payload->data;
+    payload_data = ((uint8_t *)payload->data) + sizeof(ng_udp_hdr_t);
+    udp_hdr->src_port = byteorder_htons(src_port);
+    udp_hdr->dst_port = byteorder_htons(dst_port);
+    udp_hdr->checksum = byteorder_htons(checksum);
+    udp_hdr->length = byteorder_htons(sizeof(ng_udp_hdr_t) + data_len);
+
+    memcpy(payload_data, data, data_len);
+
+    return payload;
+}
+
+static ng_pktsnip_t *build_udp_recv_exp_packet(const ng_ipv6_addr_t *src,
+        uint16_t src_port, const ng_ipv6_addr_t *dst, uint16_t dst_port,
+        uint16_t checksum, char *data, size_t data_len)
+{
+    ng_pktsnip_t *ipv6, *udp, *payload;
+    ng_udp_hdr_t *udp_hdr;
+
+    ipv6 = build_ipv6_recv_header(src, dst, data_len);
+
+    if (ipv6 == NULL) {
+        return NULL;
+    }
+
+    udp = ng_udp_hdr_build(ipv6, (uint8_t *)&src_port, sizeof(src_port),
+                           (uint8_t *)&dst_port, sizeof(dst_port));
+
+    if (udp == NULL) {
+        /* no need to release, we reset packet buffer on fail anyways */
+        return NULL;
+    }
+
+    udp_hdr = udp->data;
+    udp_hdr->checksum = byteorder_htons(checksum);
+    udp_hdr->length = byteorder_htons(sizeof(ng_udp_hdr_t) + data_len);
+
+    payload = ng_pktbuf_add(udp, data, data_len, NG_NETTYPE_UNDEF);
+
+    return payload;
+}
+
+static void test_send_ipv6(void)
```

Comment:
Would you be so kind and provide some function descriptions? At least for the test cases.
Interesting topics are classes of input values, pre- and post conditions of the test case.
It supports understanding (maybe not only for me) and eases verifying against code-coverage later.